nautobot/core/management/commands/generate_test_data.py

```diff
@@ -101,178 +111,135 @@ def _generate_factory_data(self, seed, db_name):
-        TenantFactory.create_batch(10, has_tenant_group=True, using=db_name)
-        self.stdout.write("Creating LocationTypes...")
-        LocationTypeFactory.create_batch(7, using=db_name)  # only 7 unique LocationTypes are hard-coded presently
-        self.stdout.write("Creating Locations...")
+        _create_batch(TagFactory, 15, description="on some content-types")
+        _create_batch(ContactFactory, 20)
+        _create_batch(TeamFactory, 20)
+        _create_batch(TenantGroupFactory, 10, description="without parents", has_parent=False)
+        _create_batch(TenantGroupFactory, 10, description="with parents", has_parent=True)
+        _create_batch(TenantFactory, 10, description="without a parent group", has_tenant_group=False)
+        _create_batch(TenantFactory, 10, description="with a parent group", has_tenant_group=True)
+        _create_batch(LocationTypeFactory, 7)  # only 7 unique LocationTypes are hard-coded presently
         # First 7 locations must be created in specific order so subsequent objects have valid parents to reference
-        LocationFactory.create_batch(7, has_parent=True, using=db_name)
-        LocationFactory.create_batch(40, using=db_name)
-        LocationFactory.create_batch(10, has_parent=False, using=db_name)
-        self.stdout.write("Creating Controller with Groups...")
-        ControllerFactory.create_batch(1)
-        ControllerManagedDeviceGroupFactory.create_batch(5)
-        self.stdout.write("Creating RIRs...")
-        RIRFactory.create_batch(9, using=db_name)  # only 9 unique RIR names are hard-coded presently
-        self.stdout.write("Creating RouteTargets...")
-        RouteTargetFactory.create_batch(20, using=db_name)
-        self.stdout.write("Creating Namespaces...")
-        NamespaceFactory.create_batch(10, using=db_name)
-        self.stdout.write("Creating VRFs...")
-        VRFFactory.create_batch(10, has_tenant=True, using=db_name)
-        VRFFactory.create_batch(10, has_tenant=False, using=db_name)
-        self.stdout.write("Creating VLANGroups...")
-        VLANGroupFactory.create_batch(20, using=db_name)
-        self.stdout.write("Creating VLANs...")
-        VLANFactory.create_batch(20, using=db_name)
-        self.stdout.write("Creating Prefixes and IP Addresses...")
+        _create_batch(LocationFactory, 7, description="as structure", has_parent=True)
+        _create_batch(LocationFactory, 40)
+        _create_batch(LocationFactory, 10, description="without a parent Location", has_parent=False)
+        _create_batch(ControllerFactory, 1, description="without a Device or DeviceRedundancyGroup")
+        _create_batch(ControllerManagedDeviceGroupFactory, 5, description="to contain Devices")
+        _create_batch(RIRFactory, 9)  # only 9 unique RIR names are hard-coded presently
+        _create_batch(RouteTargetFactory, 20)
+        _create_batch(NamespaceFactory, 10)
+        _create_batch(VRFFactory, 20)
```

Comment:
With or without tenants? Is it important to preserve that distinction?

Reply:
Good spot. My thinking was that `has_tenant` would naturally be a 50/50 coin flip, but looking again at the IPAM factories it's actually coded as 75% yes / 25% no (`has_tenant = factory.Faker("boolean", chance_of_getting_true=75)`). I could revert this part, but given that it's passing tests anyway with my change, I'd prefer to keep it since it's one less set of "special" calls to the factories (see also my TODO comment below on the CircuitTerminationFactory calls)